diff --git a/cmake/Open3DAddEncodedShader.cmake b/cmake/Open3DAddEncodedShader.cmake
index 36bcbb83..edb5e28e 100644
--- a/cmake/Open3DAddEncodedShader.cmake
+++ b/cmake/Open3DAddEncodedShader.cmake
@@ -65,6 +65,7 @@ function(open3d_add_encoded_shader target)
     )
 endfunction()
 
+set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64")
 # Helper target for open3d_add_encoded_shader
 if (NOT TARGET ShaderEncoder)
     add_executable(ShaderEncoder EXCLUDE_FROM_ALL)
